apps/web/components/dashboard/sidebar-pinboard.tsx

```diff
@@ -74,27 +145,173 @@ const PinboardListButton = forwardRef<
+    <div
+      ref={ref}
+      className="group/item flex justify-center space-y-1"
+      {...rest}
+    >
       <Button
         asChild
         variant="ghost"
         className={cn(
           'w-full',
           item.slug === segment ? 'bg-accent' : '',
           isCollapsed()
-            ? 'inline-flex w-auto items-center justify-center px-[10px]'
+            ? 'flex w-auto items-center justify-center px-[10px]'
             : 'inline-block justify-start overflow-hidden overflow-ellipsis whitespace-nowrap px-4 py-2'
         )}
       >
-        <Link href={item.href}>
-          <span className={cn('text-base', isCollapsed() ? 'pr-0' : 'pr-1')}>
-            {item.icon}
-          </span>
-          {!isCollapsed() && <span>{item.name}</span>}
-        </Link>
+        <div className="flex w-auto flex-row items-center justify-between">
+          <div className="flex flex-row items-center">
+            <Popover open={emojiPopoverOpen} onOpenChange={setEmojiPopoverOpen}>
+              <PopoverTrigger>
+                <span
+                  className={cn(
+                    'text-center text-base',
+                    isCollapsed() ? 'pr-0' : 'pr-2'
+                  )}
+                >
+                  {item.icon}
+                </span>
+              </PopoverTrigger>
+              <PopoverContent side="right" className="w-fit">
+                <EmojiPicker
+                  style={{ border: 'none' }}
+                  onEmojiClick={async obj => {
+                    await updateListEntry({ icon: obj.emoji });
+                  }}
+                />
+              </PopoverContent>
+            </Popover>
+            <Link href={item.href}>
+              {!isCollapsed() && (
+                <span className="block max-w-[110px] overflow-hidden text-ellipsis group-hover/item:w-[80px]">
+                  {item.name}
+                </span>
+              )}
+            </Link>
+          </div>
+          <DropdownMenu open={popoverOpen} onOpenChange={setPopoverOpen}>
+            <DropdownMenuTrigger
+              className={cn('invisible', {
+                'group-hover/item:visible': !isCollapsed(),
+                'hidden w-0': isCollapsed(),
+              })}
+            >
+              <MoreHorizontal className="h-4 w-4" />
+            </DropdownMenuTrigger>
+            <DropdownMenuContent side="right">
+              <Command>
+                <CommandGroup>
+                  <CommandItem>
+                    <Input
+                      className="h-8"
+                      value={listName}
+                      onChange={event => setListName(event.target.value)}
+                      onKeyDown={async event => {
+                        if (event.code === 'Enter') {
+                          event.preventDefault();
+                          updateListEntry({ name: listName });
+                        }
+                      }}
+                    />
```

Comment:
https://github.com/keeep-app/keeep/assets/46243719/6751812a-53e4-423e-976b-fde5d6972f58

This behavior wasn't immediately apparent to me. I had assumed that you can just type, close the dropdown, and the title gets updated. Perhaps we can add a save button or automatically update the title on change (with a debounce)?